api/commands/state.ts

```diff
@@ -8,11 +8,19 @@ export async function state(
   const allTables = await Promise.all(
     compiledGraph.tables.map(async t => dbadapter.table(t.target))
   );
-
   // filter out tables that don't exist
   const tablesWithValues = allTables.filter(table => {
     return !!table && !!table.type;
   });
 
-  return { tables: tablesWithValues };
+  let cachedStates: dataform.IPersistedTableMetadata[] = null;
+
+  if (compiledGraph.projectConfig.useRunCache) {
+    try {
+      cachedStates = await dbadapter.persistedStateMetadata();
+    } catch (err) {
+      cachedStates = [];
```

Comment:
leave a comment, clarifying that the worst case is that we were unable to load cached state so this is treated equivalently to the cache being empty?

Reply:
Added